test/readonly_writeonly.test.js

```diff
@@ -67,17 +67,26 @@ test('removing readOnly prop even if keeping', function (assert) {
   assert.deepEqual(result, expected, 'readOnly prop removed')
 })
 
-test('removing writeOnly prop', function (assert) {
+test('removing writeOnly prop & required', function (assert) {
   assert.plan(1)
 
   var schema = {
     type: 'object',
+    required: ['prop1', 'prop2', 'prop3', 'prop4'],
```

Comment:
I see a lot of code style changes for the better, but I also see you are changing some logic in this test? Could you explain what is going on here please?